src/gui/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+set(gui_SRCS
+	${CMAKE_CURRENT_SOURCE_DIR}/guiChatConsole.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/guiEditBoxWithScrollbar.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/guiEngine.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/guiFormSpecMenu.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/guiKeyChangeMenu.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/guiPasswordChange.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/guiPathSelectMenu.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/guiTable.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/guiVolumeChange.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/intlGUIEditBox.cpp
+	${CMAKE_CURRENT_SOURCE_DIR}/touchscreengui.cpp
```

Comment:
this should not be compile for non-android
also note that you need to update `build/android/jni/Android.mk`

Reply:
Is *this* `touchscreengui.cpp`? I just noticed that it doesn’t compile at all. So I removed it from the list.
But note that some x86 devices (like my laptop) have touchscreens as well, so that should not be Android-specific.

Reply:
> so that should not be Android-specific.

in theory yes
in practice the desktop "version" of irrlicht does not support touch events at all (which is why it won't build)